Remove TP note from section Integrating Satellite with Red Hat Single…

suyogsainkar opened the pull request:
… Sign-On for External Authentication

[RFE] Remove TP from Integrating Satellite with Red Hat Single Sign On
https://bugzilla.redhat.com/show_bug.cgi?id=1827082

rabajaj0509 commented:
I am not sure if we should remove this right now or not, let me confirm and get back. Till then please don't merge this one :)

suyogsainkar commented:
@rabajaj0509 Understood, I'll wait for your feedback. Note that the updates in this PR/BZ are aimed at 6.8 GA, so any updates tested/verified/merged will be published at GA. Thanks.

rabajaj0509 commented:
@suyogsainkar @spetrosi there are a few bits and pieces remaining for the documentation to be complete, do we still want to merge this one and then talk about the minor bit and pieces or we want to get them merged first and then merge this one. I leave the option up to you folks.

Few changes required in the Documentation are:

a) In the hammer section we need to change the `--app-name` to `hammer-openidc` like
```
keycloak-httpd-client-install --app-name hammer-openidc \
 --keycloak-server-url " {Keycloak-short}.example.com_" \ 
 --keycloak-admin-username "_{Keycloak-short}_User_" \
 --keycloak-realm "_{Keycloak-short}_Realm_" \ 
 --keycloak-admin-realm master \ 
 --keycloak-auth-role root-admin  \ 
 -t openidc -l /users/extlogin --force
```
b) We have not added information of handing groups in the Satellite side while we have covered how it is done on the RHSSO side of the documentation.
https://drive.google.com/file/d/1EGwoFsBRJWGGNT6ERyylJnriS6BqblnG/view?usp=sharing

suyogsainkar commented:
@rabajaj0509 thanks for taking a look and sharing the above info. I'll look into the above feedback items a. and b. and push additional commit(s)/updates to the same PR and ping you for another review. We can get the PR merged after all of the items are reviewed and verified (as part of our workflow, once the content is approved for technical accuracy by dev/qe, @spetrosi will peer review and approve/merge the PR). Thanks.

rabajaj0509 reviewed (approved):
Once we are done with the other 2 PR, we can merge this, approving this PR.

suyogsainkar commented:
@rabajaj0509 i've added steps about user and group by referring to the demo. that said, I wasn't sure about the placement of these steps, need clarification on the workflow. also a couple of existing steps such as step 7.vi. didn't match with the UI/your demo. I'll ping you on chat. thanks.

suyogsainkar commented:
@rabajaj0509 Hi Rahul, thanks for meeting earlier today. As discussed, I've made those adjustments to the steps. I've also grouped the steps performed on keycloak vs foreman UI together for better flow and readability. Also, since BZ 1792131 is now resolved, I've updated step 16 about setting organization and location. Let me know what you think.

omkarkhatavkar commented:
Set the value for the Open IDC audience:
```
# hammer settings set --name oidc_audience \
--value "['{satellite-example-com}']"
```

1. Regarding mentioned point at 14/above we should not say set the audience, instead we should say add/insert the hammer audience 
 
Add the value for the Hammer Client in Open IDC audience:
```
# hammer settings set --name oidc_audience \
--value "['{satellite-example-com-foreman-openidc}', '{satellite-example-com-hammer-openidc}']" 
```
2. Regarding point 19 It should be '{satellite-example-com-foreman-openidc}' not only ''{satellite-example-com}''

please take a look at **https://access.redhat.com/documentation/en-us/red_hat_satellite/6.7/html/administering_red_hat_satellite/chap-red_hat_satellite-administering_red_hat_satellite-configuring_external_authentication#integrating-satellite-with-red-hat-single-sign-on-for-external-authentication**

rabajaj0509 commented:
> Set the value for the Open IDC audience:
> 
> ```
> # hammer settings set --name oidc_audience \
> --value "['{satellite-example-com}']"
> ```
> 
>     1. Regarding mentioned point at 14/above we should not say set the audience, instead we should say add/insert the hammer audience keeping existing satellite ui audience

I don't have a strong opinion on this one, I will leave it up to you folks.

> 
> 
> Add the value for the Hammer Client in Open IDC audience:
> 
> ```
> # hammer settings set --name oidc_audience \
> --value "['{satellite-example-com-foreman-openidc}', '{satellite-example-com-hammer-openidc}']" 
> ```

If a user is not using the UI, then this command would not be needed, therefore, we might want to add a note, something like:
If you have already registered another client with keycloak, then make sure to append that audiance in the array too. Fore example:
 ```
 # hammer settings set --name oidc_audience \
 --value "['{satellite-example-com-foreman-openidc}', '{satellite-example-com-hammer-openidc}']" 
 ```


> 
>     1. Regarding point 19 It should be '{satellite-example-com-foreman-openidc}' not only ''{satellite-example-com}''
> 
> 
> please take a look at **https://access.redhat.com/documentation/en-us/red_hat_satellite/6.7/html/administering_red_hat_satellite/chap-red_hat_satellite-administering_red_hat_satellite-configuring_external_authentication#integrating-satellite-with-red-hat-single-sign-on-for-external-authentication**
Good catch!

rabajaj0509 reviewed (changes requested):
Almost there, just comments and we should be good to go.

suyogsainkar commented:
@omkarkhatavkar @rabajaj0509 thanks for your review and comments! I've applied your suggestions in my latest commit - please review.

rabajaj0509 commented on guides/doc-Administering_Red_Hat_Satellite/topics/proc_integrating-satellite-with-red-hat-single-sign-on-for-external-authentication.adoc:
```diff
@@ -218,12 +217,12 @@ For more information about group mappers, see https://access.redhat.com/document
 . Open the `_{Keycloak-short}.example.com_/auth/realms/_{Keycloak-short}_REALM_/.well-known/openid-configuration` URL and note the values to populate the options in the following steps.
 +
-. Set the value for the Open IDC audience:
+. Add the value for the Hammer client in the Open IDC audience. Note that if you have already registered another client with keycloak, make sure also to append that audience in the array. For example:
```
I would keep the earlier version  and add this as a separate note below.

suyogsainkar replied:
Ack, done.

rabajaj0509 commented on guides/doc-Administering_Red_Hat_Satellite/topics/proc_integrating-satellite-with-red-hat-single-sign-on-for-external-authentication.adoc:
```diff
@@ -82,39 +71,49 @@ For more information about audience support, see https://access.redhat.com/docum
 .. Click *Save*.
-
-. In the {Project} web UI, navigate to *Administer* > *Settings*, and click the *Authentication* tab.
+.. Navigate to *Users* and click *Add user*. 
```
I think, it would be better if we created the group before the user, so that user can add itself into a group directly. Else, here the admin is creating a user, then creating a group and then again going to the user to add it into a group. Does it make sense? Let me know your thoughts.

suyogsainkar replied:
Makes sense. Swapped the steps to create group and user. Does it look right?

omkarkhatavkar reviewed (approved):
ACK, from my end

rabajaj0509 commented on guides/doc-Administering_Red_Hat_Satellite/topics/proc_integrating-satellite-with-red-hat-single-sign-on-for-external-authentication.adoc:
```diff
@@ -218,14 +218,24 @@ For more information about group mappers, see https://access.redhat.com/document
 [options="nowrap", subs="+quotes,attributes"]
 ----
 # hammer settings set --name oidc_audience \
 --value "['_{foreman-example-com}_']"
```
This should be:
```
hammer settings set --name oidc_audience \
--value "['_{foreman-example-com}_-hammer-openidc']"
```

suyogsainkar replied:
Fixed, thanks.

rabajaj0509 commented on guides/doc-Administering_Red_Hat_Satellite/topics/proc_integrating-satellite-with-red-hat-single-sign-on-for-external-authentication.adoc:
```diff
@@ -82,39 +71,50 @@ For more information about audience support, see https://access.redhat.com/docum
+. Click *Create User Group*, and enter a *Name* for the user group.
+. Click the *Roles* tab, and assign appropriate roles to the user group.
+. Click the *External Groups* tab, then click *Add external user group*.
+. Enter the name of the external user group you created in the {Keycloak} web UI.
```
I see a lot of reference using `you`. Is that fine? I am not sure, you will have a better take on this one.

suyogsainkar replied:
Reworded.

rabajaj0509 reviewed (changes requested):
Almost there :smile: 

Left a few inline comments for you :+1:

rabajaj0509 commented on guides/doc-Administering_Red_Hat_Satellite/topics/proc_integrating-satellite-with-red-hat-single-sign-on-for-external-authentication.adoc:
```diff
@@ -82,39 +71,50 @@ For more information about audience support, see https://access.redhat.com/docum
++
+* Click *Save*. The user is now created.
++
+* Click the *Credentials* tab and reset the user password if required. 
```
nit: Resetting the password is not a optional step, passwords needs to be set and the `Temporary` must be set to off.

suyogsainkar replied:
Fixed, thanks.

rabajaj0509 reviewed (changes requested):
just one tiny comment :medal_sports:

rabajaj0509 commented:
Also, can you please squash all the commits to one before we merge?

suyogsainkar commented:
@omkarkhatavkar @rabajaj0509 thanks again for your thorough reviews. I've addressed all the feedback items. 
@rabajaj0509 btw, once you've approved the PR, @spetrosi will peer review and merge the PR using Github's Squash and merge button. thanks.

rabajaj0509 reviewed (approved):
Great job, really appreciate your efforts! @suyogsainkar 

Your 6th commit has tests failing in it, I would not suggest a rebase and merge in this case, would you please be willing to squash the commits before we merge the PR?

suyogsainkar commented:
Ack, @rabajaj0509 thanks.

spetrosi commented on guides/doc-Administering_Red_Hat_Satellite/topics/proc_integrating-satellite-with-red-hat-single-sign-on-for-external-authentication.adoc:
```diff
@@ -82,39 +71,50 @@ For more information about audience support, see https://access.redhat.com/docum
 +
 . Locate the *OIDC Algorithm* row, and in the *Value* column, set the algorithm for encoding on {Keycloak}, for example, `RS256`.
-. Locate the *OIDC Audience* row, and in the *Value*  column, set the value to the client ID for {Keycloak}: `['_{foreman-example-com}_']`.
+. Locate the *OIDC Audience* row, and in the *Value*  column, set the value to the client ID for {Keycloak}: `['_{foreman-example-com}_-foreman-openidc}_']`.
```
This expands to `['satellite.example.com-foreman-openidc}_']`, seems like a curly bracket is missing.

suyogsainkar replied:
Fixed. The second curly bracket was in fact not required, so removed it.

spetrosi commented on guides/doc-Administering_Red_Hat_Satellite/topics/proc_integrating-satellite-with-red-hat-single-sign-on-for-external-authentication.adoc:
```diff
@@ -82,39 +71,50 @@ For more information about audience support, see https://access.redhat.com/docum
-# hammer auth-source external update --id _Authentication Source ID_ \
---location-ids _Location ID_ --organization-ids _Organization ID_
-----
+. To set the organization and location for the authentication source:
```
```suggestion
. To set the organization and location for the authentication source, complete the following steps:
```
Use full sentence

suyogsainkar replied:
Fixed

spetrosi commented on guides/doc-Administering_Red_Hat_Satellite/topics/proc_integrating-satellite-with-red-hat-single-sign-on-for-external-authentication.adoc:
```diff
@@ -218,13 +218,23 @@ For more information about group mappers, see https://access.redhat.com/document
 +
+[NOTE]
+====
+If you have already registered another client with keycloak, make sure also to append that audience in the array. For example:
```
```suggestion
If you register several {Keycloak} clients to {Project}, ensure to append all audiences in the array. For example:
```
1. Use the attribute {Keycloak}.
2. Use one-word `ensure` instead of the phrasal verb `make sure`.
3. Remove also.
4. Use a present tense.

suyogsainkar replied:
"make sure to" is a set phrase. "ensure to" is incorrect. I've reworded slightly and applied the change.

spetrosi commented on guides/doc-Administering_Red_Hat_Satellite/topics/proc_integrating-satellite-with-red-hat-single-sign-on-for-external-authentication.adoc:
```diff
@@ -242,13 +252,14 @@ For more information about group mappers, see https://access.redhat.com/document
 ----
 +
-. Until https://bugzilla.redhat.com/show_bug.cgi?id=1792131[BZ#1792131] is resolved, you must use the Hammer CLI to set the organization and location. To set the organization and location, you must first retrieve the ID of the {Keycloak} authentication source:
+. To set the organization and location of the authentication source:
```
```suggestion
. To set the organization and location of the authentication source, complete the following steps:
```
Use the full sentence for clarity.

suyogsainkar replied:
Fixed

spetrosi commented on guides/doc-Administering_Red_Hat_Satellite/topics/proc_integrating-satellite-with-red-hat-single-sign-on-for-external-authentication.adoc:
```diff
@@ -242,13 +252,14 @@ For more information about group mappers, see https://access.redhat.com/document
 +
-. Until https://bugzilla.redhat.com/show_bug.cgi?id=1792131[BZ#1792131] is resolved, you must use the Hammer CLI to set the organization and location. To set the organization and location, you must first retrieve the ID of the {Keycloak} authentication source:
+. To set the organization and location of the authentication source:
+.. First, retrieve the ID of the {Keycloak} authentication source.
```
```suggestion
.. Retrieve the ID of the {Keycloak} authentication source:
```
Remove fluff

suyogsainkar replied:
Done

spetrosi commented on guides/doc-Administering_Red_Hat_Satellite/topics/proc_integrating-satellite-with-red-hat-single-sign-on-for-external-authentication.adoc:
```diff
@@ -242,13 +252,14 @@ For more information about group mappers, see https://access.redhat.com/document
 ----
 +
-. Set the location and organization:
+.. Set the location and organization.
```
```suggestion
.. Set the location and organization:
```
Throughout Satellite documentation, we use colons at the end of steps that are followed by a command.

suyogsainkar replied:
Ack, noted.

spetrosi commented on guides/doc-Administering_Red_Hat_Satellite/topics/proc_integrating-satellite-with-red-hat-single-sign-on-for-external-authentication.adoc:
```diff
@@ -218,13 +218,23 @@ For more information about group mappers, see https://access.redhat.com/document
 . Open the `_{Keycloak-short}.example.com_/auth/realms/_{Keycloak-short}_REALM_/.well-known/openid-configuration` URL and note the values to populate the options in the following steps.
 +
-. Set the value for the Open IDC audience:
+. Add the value for the Hammer client in the Open IDC audience. 
```
```suggestion
. Add the value for the Hammer client in the Open IDC audience:
```

suyogsainkar replied:
Done

suyogsainkar commented:
@spetrosi thanks for your review and comments. I've applied your suggestions in my latest commit.
